From 5e691a06a2be210ca681a9173c076c9ce1df177c Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Wed, 20 Dec 2023 11:08:58 +0100
Subject: [PATCH 0558/1695] dts: stm32: add Cortex-M33 remoteproc node in
 stm32mp251.dtsi

Introduces the m33_rproc node under the mlahb simple bus.
This declaration is a aligned with the Linux DT.
The mlahb bus provides memory translation between
the Cortex-M and Cortex-A memory mappings.
For the STM32MP2, no memory translation is needed, and the
dma-ranges is declared accordingly.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 36738acbc1072ae104e7e734c04c29b2da038fd1 dts: stm32: add Cortex-M33 remoteproc node in stm32mp251.dtsi]
Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: I8eb09cbf9f466bcbabc039c4c0663e0227738146
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/358618
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 core/arch/arm/dts/stm32mp251.dtsi | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index b9ebf936c..0cb3ce4b6 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -85,6 +85,23 @@
 		};
 	};
 
+	mlahb: ahb@1 {
+		compatible = "st,mlahb", "simple-bus";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges = <0x0 0x0  0xFFFFFFFC>;
+		dma-ranges = <0x0 0x0 0xFFFFFFFC>;
+
+		m33_rproc: m33@0 {
+			compatible = "st,stm32mp2-m33";
+			reg = <0 0>;
+			resets = <&rcc C2_R>,
+				 <&rcc C2_HOLDBOOT_R>;
+			reset-names = "mcu_rst", "hold_boot";
+			status = "disabled";
+		};
+	};
+
 	scmi_regu: scmi-regu {
 		compatible = "st,scmi-regulator-consumer";
 		#address-cells = <1>;
-- 
2.47.2

